### PR TITLE
UWP Updates

### PR DESCRIPTION
**Pull request by mattleibow**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,24 +1,277 @@
 .xpkg/
-bin
```
Comment (mattleibow) [inline .gitignore]: Using the .gitignore from GitHub - correctly removes the VS temp files

File: component.yaml [hunk 2]
```diff
@@ -56,3 +56,13 @@ samples:
     path: samples/Mac/GravatarMobileMac.sln
     configuration: "Release"  
     nobuild: true
+
+# XAML files require extra resources
+additional-files:
+ - src: src/compiled/GravatarMobile.UWP.pri
+   dst: lib/uap/GravatarMobile.UWP.pri
+ - src: src/compiled/GravatarMobile.UWP/GravatarMobile.UWP.xr.xml
+   dst: lib/uap/GravatarMobile.UWP/GravatarMobile.UWP.xr.xml
```
Comment (mattleibow) [inline component.yaml]: UWP libraries with XAML files store XAML resources outside the assembly

File: component.yaml [hunk 3]
```diff
@@ -56,3 +56,13 @@ samples:
...
+
+# XAML files require extra resources
+additional-files:
+ - src: src/compiled/GravatarMobile.UWP.pri
+   dst: lib/uap/GravatarMobile.UWP.pri
+ - src: src/compiled/GravatarMobile.UWP/GravatarMobile.UWP.xr.xml
+   dst: lib/uap/GravatarMobile.UWP/GravatarMobile.UWP.xr.xml
+# workaround for bug in xamarin-component
+ - src: samples/UWP/GravatarMobile_UWP/project.json
+   dst: samples/GravatarMobile-UWP/GravatarMobile_UWP/project.json
```
Comment (mattleibow) [inline component.yaml]: there seems to be a bug with xamarin-component, I will have a look now...

File: samples/UWP/GravatarMobile-UWP.sln [hunk 4]
```diff
@@ -11,17 +11,14 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "GravatarMobile.UWP", "..\..
 EndProject
 Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
-		Debug|Any CPU = Debug|Any CPU
```
Comment (mattleibow) [inline samples/UWP/GravatarMobile-UWP.sln]: Any CPU is not a valid configuration for UWP apps - it gets compiled to native code

Comment (mattleibow) [inline samples/UWP/GravatarMobile-UWP.sln]: I removed it as VS selects this option as the default. If the user opens the solution, it defaults to this and results in a build error.

File: component.yaml (hunk 3 above)
Comment (mattleibow) [inline component.yaml]: https://github.com/xamarin/xamarin-component/pull/57

**Comment by newky2k** [comment]
Merging changes from @mattleibow  to make UWP work

**Pull request merged**